decidim-initiatives/app/controllers/decidim/initiatives/create_initiative_controller.rb

```diff
@@ -26,112 +25,89 @@ class CreateInitiativeController < Decidim::Initiatives::ApplicationController
       helper_method :promotal_committee_required?
 
       before_action :authenticate_user!
+      before_action :ensure_type_exists, only: [:previous_form, :fill_data, :show_similar_initiatives, :promotal_committee, :finish]
+      before_action :ensure_initiative_exists, only: [:fill_data, :show_similar_initiatives, :promotal_committee, :finish]
 
-      steps :select_initiative_type,
-            :previous_form,
-            :show_similar_initiatives,
-            :fill_data,
-            :promotal_committee,
-            :finish
+      def select_initiative_type
+        redirect_to previous_form_create_initiative_index_path if single_initiative_type?
+        @form = form(Decidim::Initiatives::SelectInitiativeTypeForm).from_params(params)
 
-      before_action :ensure_type_exists, only: :show
+        render :select_initiative_type && return unless request.put?
```

Comment:
This seems like we're hacking Rails a bit, using the same method for what it would be two methods (`new` and `create` in the typical Rails Way). This is what we have right now: 

```ruby
# GET /create_initiative/select_initiative_type
# PUT /create_initiative/select_initiative_type
def select_initiative_type
```

(It's the same in the other methods too)

I understand that you want to keep the same URLs for not introducing too much changes compared to the current behavior.

Personally, I think that the current URLs are a bit messy (like having `/create_initiative/select_initiative_type` instead of `/initiatives/new` or having  `/initiatives/i-:id` instead of  `/initiatives/:id` or `/initiatives/:slug`). We should discuss and plan a better model for this, probably similar to what we have in Proposals that sounds a bit more "railsy", but that's out of the scope of the current PR.

So, having this in mind, I'd like to at least separate the two HTTP methods. The problem is that  I don't have a semantic naming yet, but maybe by talking about this we can reach to something. 

How about something like this? 

```ruby
# GET /create_initiative/select_initiative_type
def select_initiative_type 

# PUT /create_initiative/select_initiative_type
def create_select_initiative_type 
```

As I said, I'm not 100% on-board with this change, but I'd like to at least open the discussion.

Reply:
I do not have strong feeling about this change. Ideally we should have more controllers, each one to implement new / create methods.

Reply:
I have implemented something in this direction.

Reply:
LGTM, thanks!